engine/src/configuration/configuration.h

```diff
@@ -0,0 +1,95 @@
+#ifndef CONFIGURATION_H
+#define CONFIGURATION_H
+
+
+struct Configuration
+{
+    // Config Structs Declaration
+
+    // These are settings that are relevant to collision.cpp and are marked as hacks
+    // or their comments are unclear or they are unclear
+    struct CollisionHacks {
+        const float collision_hack_distance;
```

Comment:
I would suggest we put this in the `VegaStrike::Configuration` or `VegaStrikeEngine::Configuration` namespace, and then have all the structures in parallel there. It's cleaner overall and easier to verify the structures that way.

The actual global configuration object instance can be in a different namespace if desired.

Reply:
Personally, I think this is fine the way it is.

Reply:
So, to separate the two comments:

1) From the documentation on gtest, I suspect we'd be well served to make a habit of putting things in namespaces, because it will make subsequent test code easier to write

2) I'd prefer to not have the other structs declared within the top level just to make the top level config class a bit easier to read, especially since it will be the only one that will eventually become a class with getter/setter methods, rather than a naked public struct. It's not a blocking issue for me though.

Reply:
1. Oh. Yes, let's put this in a namespace. I didn't notice until now that this code was not in any namespace at all.
2. We can separate out the other structs if you guys insist. Perhaps that is better.

My primary goal here is just to get this PR approved, passing, and merged. It's been sitting here long enough, IMHO.

To that end, perhaps the CTest additions can be omitted for now? It might be better to figure them out in another PR.

Reply:
separating out the testing into its own PR might be a good idea

Reply:
I'm going to resolve this discussion with the following:
1. I'll add a namespace and move the structs outside.
2. I'll keep the tests, but drop google test for now. 
3. Where the call is made from a nested loop and there are performance implications, it can be placed in a static local variable, which will only be called once. The person refactoring that code will make that decision.